src/lib/autoinstall/autosetup_helpers.rb

```diff
@@ -159,7 +152,40 @@ def autosetup_network
     # @return [Boolean] true when network config should be written before the
     #   proposal; false when not
     def network_before_proposal?
-      !!@network_before_proposal
+      return @network_before_proposal unless @network_before_proposal.nil?
+
+      @network_before_proposal = networking_section.fetch("setup_before_proposal", false)
+    end
+
+    # Defines convenience methods to check the existence and the content of
+    # the current {Yast::Profile} sections
+    #
+    # @example networking section methods
+    #
+    #   profile = { "networking" => { "setup_before_proposal" => true } }
+    #   Yast::Profile.current = profile
+    #
+    #   client.networking_section? #=> true
+    #   client.networking_section["setup_before_proposal"] #=> true
+    #   client.not_present_section? #=> false
+    def method_missing(method, *arguments, &block)
```

Comment:
Probably it is preferable to add the methods as needed and just defined networking and general section methods by now... or at least only respond_to supported sections missing methods.

@imobachgs , @schubi2  what do you think?

Reply:
I am not a big fan of `method_missing` (although it may have legit use cases). In this case, I would rather prefer to add the needed methods (as you have proposed) and start tthinking about turning `Yast::Profile` into a proper class with a better API.

Reply:
Alternatively, I would add those methods (#section? and #section) to the `Yast::Profile` module by now.

Reply:
agree with first comment, about the second one, I still prefer the convenience methods, I mean calling something like `general_section?` instead of `Yast::Profile.section?("general")`

Reply:
> Um, to be honest I'm not a big fan of metaprogramming, it should be used only when it is really needed or when it helps to make the code generic.
> 
> The problem is that you cannot easily find the place where the code is defined, you cannot simply grep the sources for the method name. E.g. grepping for `networking_section` whole YaST sources will not find any place. (OK, it would find the comment above wink, but for the others it would not work... stuck_out_tongue)

Yep, in general I think here it doesn't make sense to use it at all.

> 
> In this case it can be easily implemented as a regular method
> 
> ```ruby
> def section_present?(name)
>   Yast::Profile.current.keys.include?(name)
> end
Also agree, something like  `section_present?(name)` and `section_content(name)` should be more than enough.

> ```
> 
> Or is there any advantage of `method_missing` which I overlooked?

Don't think so ;)

Reply:
> But I prefered to use
> 
> ```ruby
> general_section["semi-automatic"]
> network_section["setup_before_proposal"]
> ```
> 
> instead of
> 
> ```ruby
> section("general")["semi-automatic"]
> section("network")["setup_before_proposal"]
> ```

Alternatively we could use splat arguments and return either the whole section or just a single value, e.g.

```ruby
section("general") # =>  Hash
section("general", "semi-automatic") # => Boolean
```